packages/utilities/src/merge.test.ts

```diff
@@ -87,4 +87,23 @@ describe('merge', () => {
       foo: { bar: 'baz' },
     });
   });
+
+  it('can handle arrays', () => {
```

Comment:
add a case to verify what happens when you mix and array in target with any of these values in source:

* null/undefined (does it clear the array? I *think so*)
* string, number, boolean (does it append the value to the array or replace? I think it replaces the array.)
* object (does it replace the array? I think replace)

Reply:
Updated various tests and removed concat option since I think it could negatively impact scenarios like props.children.